python/Auth0/main.py

```diff
@@ -29,63 +34,92 @@ def lookahead(self, iterable):
             last = val
         yield last, False
 
+
+    async def refresh_token(self):
+        response = requests.post(
+                    self.api_url, headers=TOKEN_HEADER, data=self.token_payload
+                    )
+        if response.status_code == TokenRefreshResult.SUCCESS.value:
+            self.logger.info("Auth0: Token succesfully refreshed!")
+            return TokenRefreshResult.SUCCESS.value, json.loads(response.text)["access_token"]
+        
+        self.logger.error("Auth0: Refreshing the token has failed!")
+        return TokenRefreshResult.FAILURE.value, None
+        
+
     async def get_auth(self):
-        last_verified = ""
-        last_user = ""
+
+        code, token = await self.refresh_token()
+
+        if code != 200:
+            raise Exception(f"Auth0: Refreshing the token has failed!")
+        
+        self.logger.info(f"Auth0: Succesfully initiated the token - code: {code}, token: {token}")
+
+        last_verified = str()
+        last_user = str()
         buffer_verified = ShiftArray()
         buffer_users = ShiftArray()
-        response = requests.post(
-            self.api_url, headers=self.token_header, data=self.request_data
-        )
-        token = json.loads(response.text)["access_token"]
+
         while True:
             headers = {"authorization": f"Bearer {token}"}
-            verified =  await requests_async.get(self.verified_url, headers=headers)
-            users = await requests_async.get(self.users_url, headers=headers)
-            if verified.status_code != 200:
+
+            try:
+                verified =  requests.get(self.verified_url, headers=headers)
+                users = requests.get(self.users_url, headers=headers)
```

Comment:
could we also please log how many users were retrieved? (not to log any user data but to have an indication of successful user list retrival)